src/platform/react-hooks/src/utils/utils.ts

```diff
@@ -0,0 +1,36 @@
+import * as API from 'ably';
+
+export const arrEvery = (Array.prototype.every as unknown)
+  ? function <T>(arr: Array<T>, fn: (value: T, index: number, arr: Array<T>) => boolean) {
+      return arr.every(fn);
+    }
+  : function <T>(arr: Array<T>, fn: (value: T, index: number, arr: Array<T>) => boolean) {
+      const len = arr.length;
+      for (let i = 0; i < len; i++) {
+        if (!fn(arr[i], i, arr)) {
+          return false;
+        }
+      }
+      return true;
+    };
+
+export function shallowEquals(source: Record<string, unknown>, target: Record<string, unknown>) {
```

Comment:
I would have thought after v2 changes we would be able to just import utils from the core ably-js lib?

Reply:
removed this